### ISSUE TITLE
dune doesn't cleanup all .merlin files and doesn't inform user of using them

### ISSUE THREAD

**Issue by rand00** [ocaml/dune#4866]
## Expected Behavior
* I would have expected dune to inform me of using .merlin files instead of `dune ocaml-merlin` 
* dune should have removed all its own .merlin files from my source-tree, when it tried to do so
* I would have expected dune to inform me that it was about to remove .merlin files - possibly asking if it should do this

## Actual Behavior
* dune left one of its own generated .merlin files in my source-tree without informing me 
* I wasn't informed on upgrading dune, or on building my code, that it was removing my .merlin files
* I wasn't informed that dune was using this single .merlin file instead of `dune ocaml-merlin`

## Reproduction

- PR with a reproducing test: 

I can't reproduce, as it happened in my currently closed-source repository

## Specifications

- Version of `dune` (output of `dune --version`): 2.9.0
- Version of `ocaml` (output of `ocamlc --version`): 4.11.1
- Operating system (distribution and version): arch linux

**Comment by voodoos** [comment]
> I would have expected dune to inform me of using .merlin files instead of `dune ocaml-merlin`
It is a fair request but note that having custom `.merlin` files not handled is also a correct workflow. Maybe we could print a warning by default and have an option to deactivate it in the project file.

> dune should have removed all its own .merlin files from my source-tree, when it tried to do so
Yes it should, but is is hard to understand what happened without a reproduction. Note that the reproduction do not need to be from your real private source code, you can try to mimic part of it's structure in a minimal project that illustrate the issue.

My guess is that the database that Dune uses to keep track of generated `.merlin` files was manually deleted at some point (by manual removal of the `_build` dir), but this is not an easy thing to check since `.merlin` files could be user-created.

The warning discussed above when `.merlin` files are left in the source-tree could inform the user when such a thing happen.

**Comment by rand00** [comment]
Yes I think the warning is enough, as my biggest problem was the time used on debugging what was wrong with merlin after the dune upgrade - and OCaml beginners would have had a much harder time

**Comment by rand00** [comment]
When putting files on a users filesystem, one can't expect the files not be modified, deleted etc. - so dune should expect that user deleting the _build dir at some point is a probable situation
